sdk/iot/core/src/az_iot_hub_client.c

```diff
@@ -1,12 +1,137 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
+#include <stdint.h>
+
 #include "az_iot_hub_client.h"
+#include <az_precondition_internal.h>
+#include <az_result.h>
 #include <az_span.h>
 
 #include <_az_cfg.h>
 
+static const uint8_t hub_client_forward_slash = '/';
+static const uint8_t hub_client_param_separator = '&';
+static const uint8_t hub_client_null_terminate = '\0';
+
+static const az_span hub_client_api_version = AZ_SPAN_LITERAL_FROM_STR("?api-version=2018-06-30");
```

Comment:
Is there a convenient way to upgrade API version? Once the library is built, user application is not able to change it. For example, PnP is not supported by this version. But if user application decides to implement some features for PnP itself, `az_iot_hub_client_user_name_get` can not be used.

Reply:
We have had discussions about moving this and all versions for the sdk to a header file but it sounds here like you would want to be able to change it in other ways?

Reply:
I'm not sure if it is a reasonable request. Since the api-version is not a constant value, it would be flexible if user application is able to change it at run time, or maybe at compiler time.

Reply:
I can get further information about this but I believe the API version is essentially established at compile time since the functionality which the SDK is capable of won't change at runtime.
We didn't have this capability in the current SDK but maybe it's something we can think about. I'll leave that to @ericwol-msft and @jspaith maybe to chime in on.

Reply:
I think we should allow the user to modify the version. The reason is that the user is now more in control of talking to the service. This will allow them to implement new service features without SDK support.

Reply:
Noted for the future but for now will resolve